.github/workflows/vapoursynth.yml

```diff
@@ -0,0 +1,155 @@
+name: vapoursynth
+
+on: [push, pull_request]
+
+jobs:
+  clippy-rustfmt:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+        override: true
+        components: clippy, rustfmt
+    - name: Run rustfmt
+      uses: actions-rs/cargo@v1
+      with:
+        command: fmt
+        args: --all -- --check
+    - name: Run clippy on all tests
+      uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: --all --all-targets --all-features --exclude sample-plugin
+    - name: Run clippy on the sample-plugin
+      uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: --all-targets --package sample-plugin
+
+  unix-tests:
+    strategy:
+      matrix:
+        toolchain:
+         - stable-x86_64-apple-darwin
+         - stable-x86_64-unknown-linux-gnu
+         - stable-i686-unknown-linux-gnu
+        include:
+         - toolchain: stable-x86_64-apple-darwin
+           os: macOS-latest
+         - toolchain: stable-x86_64-unknown-linux-gnu
+           os: ubuntu-latest
+         - toolchain: stable-i686-unknown-linux-gnu
+           os: ubuntu-latest
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install VapourSynth on MacOS
+      if: matrix.os == 'macOS-latest'
+      run: |
+        brew install vapoursynth
+    - name: Install VapourSynth on Ubuntu 64-bit
+      if: matrix.toolchain == 'stable-x86_64-unknown-linux-gnu'
+      run: |
+        build/script.sh
+    - name: Install VapourSynth on Ubuntu 32-bit
+      if: matrix.toolchain == 'stable-i686-unknown-linux-gnu'
+      run: |
+        build/script.sh i686
+    - name: Install Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: ${{ matrix.toolchain }}
+        override: true
+    - name: Run tests
+      run: |
+        python3 build/run-tests.py
+    - name: Run sample-plugin
+      run: |
+        cd sample-plugin
+        cargo build --verbose
+        cargo run --verbose --bin test --features cfg-if \
+                  vapoursynth/vapoursynth-functions \
+                  vapoursynth/vsscript-functions
+    - name: Run doc
+      uses: actions-rs/cargo@v1
+      with:
+        command: doc
+        args: --all-features --verbose
+    - name: Copy index into the target directory
+      if: matrix.toolchain == 'stable-x86_64-unknown-linux-gnu'
+      run: |
+        cp build/index.html target/doc
+    - name: Deploy documentation
+      if: matrix.toolchain == 'stable-x86_64-unknown-linux-gnu'
```

Comment:
Could you also add a branch = master condition?

Reply:
I don't think it's necessary, but I want a double confirmation from you. 

`BASE_BRANCH: The base branch of your repository which you'd like to checkout prior to deploying. This defaults to master.`

I've taken that sentence from [here](https://github.com/JamesIves/github-pages-deploy-action#configuration-)

Reply:
Well, it should skip this step if not running on master, rather than do this step with the master branch when ran on any branch, shouldn't it?

Reply:
Hmm, since it always checkouts the master, I think it should always deploy the master even if you are on another branch. Or am I wrong?